### DISCUSSION TITLE
Suggestions based on test

### DISCUSSION THREAD

**Discussion by fhanrath** [Try/OpenGothic#216] [General]
Hello,

this weekend I was able to put in some hours into OpenGothic and I'm really impressed by how smooth this project runs and that I didn't run into any show stopping issues (I'm at chapter two as a mage, already visited the valley of mines)

## Things I noticed

There were a few things I noticed and tho they aren't big issues, I wanted to address them:

- [1] At first the controls felt a bit off. I think I adjusted after a while, but you can see it in your own [video](https://youtu.be/R9MNhNsBVQ0?t=290) between 4:51 and 4:52. It seems like the camera does make a small jump.
- [2] Regardless if I choose gothic 1 or gothic 2 controls, I still have to press left mouse + w to attack.
- [3] On my system some sounds played too fast (I'm aware of https://github.com/Try/OpenGothic/issues/117 but I think this is something else)
  - The music generally plays too fast.
  - Sheep noises are played in a staccato (instead of sheeps making their noise every few seconds, the sound is played in quick succession)
  - Voices, other sounds and video sound are perfectly fine
  - Maybe its a performance related issue? I had ~160 fps with drops to 60 and bumps to 300.
- [4] Some animals (big rats) don't attack (only when they are charging. When they stand in front of you, they wait until you hit and only attack then (waited for > 10s))
  - I'm not sure if this bug was also present in the original game.
- [5] Lurkers attack way too quick (like gothic 3 animals)
- [6] Pickaxes get destroyed when using them for digging up treasure
  - I'm not sure if that's intended or a bug. I'm just sure that in the original game one pickaxe was enough for all treasures.
- [7] Greg didn't show up at Dexter's (I did all Greg related quests)
- [8] I had the impression jumping and climbing up doesn't allow everything that was possible in vanilla (probably that's why in the [feature list](https://github.com/Try/OpenGothic/wiki/Feature-list) pull up is marked as partial)
  - E.g. I wasn't able to climb from the lantern onto the roof of Bosper's shop.
- [9] Using teleporters teleported the hero instantly, but the camera flies to the hero with a certain speed (~10s for Temple Ruins -> Khorinis City)
- [10] Sometimes when npcs accompany you they just stay at one spot and won't move after a fight
  - Speaking with them doesn't help, you have to pull some enemies to them and let them fight. After the fight the npc does move again
  - I encountered this with Lares (I had to follow him. I didn't have npcs follow me yet, so I'm not sure if this will be an issue in this case too)

I switched from steam vanilla to gog 2.7 after I encountered the infamous Lares bug (Lares doesn't give you the option to bring the ornament fragment to the water mages anymore and breaks the game.)
 - I tried to fix this by using the marvin mode + o to control Lares and then drop the fragment.
   - [11] But pressing o didn't let me control Lares or any other npc.

The version of OpenGothic I used to play was [b6ec4cc](https://github.com/Try/OpenGothic/commit/b6ec4cc34987b93913470ab204ceade810aa1541).

My system info:
- OS: Arch Linux
- KERNEL: 5.15.2-arch1-1
- CPU: AMD Ryzen 9 5950X 16-Core
- GPU: AMD Radeon RX 6900 XT (SIENNA_CICHLID, DRM 3.42.0, 5.15.2-arch1-1, LLVM 13.0.0)
- GPU DRIVER: 4.6 Mesa 21.2.5
- RAM: 64 GB

## How I can help

If you prefer to receive each issue in a seperate github issue, I'll gladly create those. I just thought that maybe not everything in my list will be relevant or maybe already known and worked on. And that it's maybe more efficient to talk about the things I've noticed here in one issue. For easier referencing I assigned each point a number.

And if I can provide more information regarding an issue or test something on my system, I'll do that.

Also I would like to be able to contribute more than just issues. I tried to find the code responsible for the controls to have a look at the Gothic 1/2 control thing ([2]). It's probably 'playercontrol.cpp', but I'm still a bit overwhelmed by the codebase and didn't find the differentiation between the two control schemes.

What I did is extend the [README](https://github.com/fhanrath/OpenGothic/blob/feature/arch-build/README.md) by instructions for building on Arch systems. If you're interested in this, I'm more than happy to create a pull request.

Thank you and have a great one!

**Comment by fhanrath** [comment]
regarding [3]:

I noticed that the video linked in https://github.com/Try/OpenGothic/issues/181 also seems to show the too fast music issue.

**Comment by Try** [comment]
Hi, @fhanrath !
On that's a lot o stuff! Thank you for your review!

1. That odd, camera was completely redone after #161, so video is not supper relevant. And at this point I lose ability to "see" the differences - too use to new implementation :)
2. Yep, can confirm: this switch is not implemented 
3. Music is completely covered in #117, basically the problem is to emulate DX7 music on top of low-level api. Good solution can be switch to third-party library, but there seems to be no such.
3.1 Voices/Sounds are "normal" *.wav sounds - so it's not related to music. Sheep sound, more specifically, is driven by animation tags - so game content tells me to play this sound all to often. Why this is not the case in original game - don't know at this point.
4. Also not sure, but can be related to `FightAlgo` and attack distance mis-interaction.
5. I think this is not the speed, but double appearance of `DEF_OPT_FRAME` tag - will quick fix in this ticket
6. Somewhat known, but hard to reproduce (FPS depended). 
Basically in treasure-object takes away your pickaxe and then adds it back by `DEF_INSERT_ITEM`. In a way this is concurrency case: animation gets terminated, before all tags a processed.
7. He will show, eventually ;) This is more on wonfix side: 
  Game script ask Greg to go to Dexters place. In vanilla, he is out of view-area, so he will simply teleport, to the place. but in OpenGothic - he casually walks to. Btw: same thing happens with ring of water meeting.
8. Can very well be: jumping is difficult to get exact mechanic, since physic engine is completely different - I'll take a look at this case
9. Seem to be recent regression, since #161 - will quick fix in this ticket
10. It's been a while since I've seen this case... In can be a script issue.
11. `marvin+o` - not implemented, but you can use the `insert `

> If you prefer to receive each issue in a seperate github issue

I'll create this - that's fine. One thing I can recommend is to create a discussion thread instead of issues, if it's less formal

**Reply by Try** [reply]
5, 9 - fixed in 9231f9f, 96877d7

**Reply by fhanrath** [reply]
Wow, you implemented already a few fixes :)

- [1] Yes, I barely noticed it after a while. But at first it seemed like the camera didn't move in a linear way.
- [6] I see. The animation was weird for me too because the hero did 4 strikes with the pickaxe instead of 3.
- [7] That's fine. But I don't think he would've showed up, because afterwards I visited him in the fields where he was still standing. He didn't have a dialog too.
- [10] Maybe. I tested the other case: I was able to pull Diego through the whole valley of mines without him stopping and not reacting. I think this happened with Lares after I was out of reach an he had to wait for me too.
- [11] Didn't think of that. But well, I think the start is the best part of Gothic II anyways. That and the addon ;)

I'll keep that in mind. I'm still new to contributing on github, so thank you for the directions!

**Reply by Nindaleth** [reply]
8 - I feel like climbing is a lot stricter in OpenGothic. I understand that it might not be possible to replicate the mechanic in a way that allows e.g. climbing the cliff next to the starting area and taking a shortcut to the Valley of Mines. However there are much less complex things that are quite hard to achieve compared to the original - e.g. leaving Khorinis through the market gate, going through the moat and then getting to the other gate. There are a few natural steps/stairs along the way that are hard to jump/climb over in OpenGothic.

**Comment by Try** [comment]
> I tried to find the code responsible for the controls to have a look at the Gothic 1/2 control thing ([2]). It's probably 'playercontrol.cpp',

You in the right place; At this pint there is only a G1 style controls and nothing else.
If you need to reach settings file: `Gothic::inst().settingsGetI("GAME","USEGOTHIC1CONTROLS")`
Also, created #214 for this issue - you ca assign yourself, if you want to work on it

**Reply by fhanrath** [reply]
Yes, I'll have a look at it.

**Comment by czarnyafgan** [comment]
I can add following: the wind effect strenght is too much ( too high ) and should be reduced by approximately 50%. Now trees behave in non-natural way ( they swing / sway too much and too fast ), so they give me a motion sickness ( I play on large 28'' inch display monitor. ) 

Regards

**Reply by Try** [reply]
Hm, for baseline I took 50% of amplitude and 200% of period of original game.
Don't know what can a good approach to find a 'good' value at this point...

**Comment by Joshin95** [comment]
hey guys, about the music there is a library created for the REGoth project:
https://github.com/libdmusic/libdmusic
It had a very decent progress on constructing the sounds based on gothic files, I know there were some bugs, but maybe this can be used as a decent base. @Try

**Reply by Try** [reply]
Hi, @Joshin95 !
I'm aware about this library and tested it in ~2019, before start to work on my own implementation. After testing got the impression, that this library is more in prototype stage and not production ready.
Downsides:
  1. dependency from thirdparty *.dll
  2. hight CPU usage (more then all things in opengothic combined)
  3. low sound quality on some tracks (Irdotar for example)

Also see no new commits since 2019 year..

**Comment by czarnyafgan** [comment]
Try to lower the amplitude by 50% and slow down the sway animation by 50%.

wt., 23 lis 2021, 21:21 użytkownik Try ***@***.***> napisał:

**Comment by czarnyafgan** [comment]
*slow down the sway animation speed by 50%

śr., 24 lis 2021, 09:31 użytkownik Jakub Kruszewski <
***@***.***> napisał:
